### PR TITLE
update bbb build instructions for stretch 3.9

### PR DESCRIPTION
**Pull request by redFrik**
big rewrite for debian stretch and sc3.9.  many things simplified due to the new gcc 6.3.0, added info about sc3-plugins, removed old wheezy etc

### CODE REVIEW

File: development/building-beagleboneblack.md [hunk 1]
```diff
@@ -5,157 +5,140 @@ title: Building From Source on BeagleBone Black
...
-4. `mkdir build && cd build`
-5. `export CC=/usr/bin/gcc-4.8`
-6. `export CXX=/usr/bin/g++-4.8`
-7. `cmake -L -DCMAKE_BUILD_TYPE="Release" -DBUILD_TESTING=OFF -DSUPERNOVA=OFF -DNOVA_SIMD=ON -DNATIVE=OFF -DSC_ED=OFF -DSC_WII=OFF -DSC_IDE=OFF -DSC_QT=OFF -DSC_EL=OFF -DSC_VIM=OFF -DCMAKE_C_FLAGS="-mfloat-abi=hard -mfpu=neon" -DCMAKE_CXX_FLAGS="-mfloat-abi=hard -mfpu=neon" ..`
+3. `git checkout 3.9`
+4. `git submodule init && git submodule update`
+5. `nano lang/LangSource/SC_TerminalClient.cpp`  #TEMP FIX for 100% sclang issue - find the first line and comment it out, add the 2nd right after
+        
+        //mTimer.cancel();
+        if (error==boost::system::errc::success) {mTimer.cancel();} else {return;}
```
Comment (patrickdupuis) [inline development/building-beagleboneblack.md]: There's a missing semi-colon at the end of this line.

File: development/building-beagleboneblack.md [hunk 2]
```diff
@@ -5,157 +5,140 @@ title: Building From Source on BeagleBone Black
...
 step1 (hardware setup)
 --
 1. connect an ethernet cable from the network router to the bbb
 2. insert the sd card and usb soundcard
 3. last connect usb power from a 5V@1A power supply
 
-step2 (login & preparations)
+step2 (update the system, install required libraries)
 --
 1. `ssh debian@beaglebone`  #from your laptop, default password is temppwd
```
Comment (patrickdupuis) [inline development/building-beagleboneblack.md]: The password should appear in a different style than the rest of the text. Maybe use markdown for code?

File: development/building-beagleboneblack.md [hunk 3]
```diff
@@ -5,157 +5,140 @@ title: Building From Source on BeagleBone Black
...
 2. `sudo passwd debian`  #change password
-3. `sudo /opt/scripts/tools/grow_partition.sh`  #expand file system
+3. `sudo /opt/scripts/tools/grow_partition.sh`  #expand file system. skip if you are running on the emmc
 4. `sudo reboot`  #and log in again with ssh
+5. `sudo apt-get update`
+6. `sudo apt-get upgrade`
+7. `sudo apt-get dist-upgrade`
+8. `sudo apt-get install libsamplerate0-dev libjack-jackd2-dev libsndfile1-dev libasound2-dev libavahi-client-dev libicu-dev libreadline-dev libfftw3-dev libxt-dev libudev-dev libcwiid-dev cmake git build-essential python-dev alsa-utils cpufrequtils`
```
Comment (patrickdupuis) [inline development/building-beagleboneblack.md]: Is it important to include `libjack-jackd2-dev` in this list? Aren't we building jack from source in the next step?

File: development/building-beagleboneblack.md [hunk 4]
```diff
@@ -5,157 +5,140 @@ title: Building From Source on BeagleBone Black
...
 
+**done!** see below for sc3-plugins, autostart, benchmarks and notes
 
 - - -
 
-wheezy (older system)
+sc3-plugins
 ==
+how to compile and install [sc3-plugins](https://github.com/supercollider/sc3-plugins).
+1. `git clone --recursive https://github.com/supercollider/sc3-plugins.git --depth 1`
```
Comment (patrickdupuis) [inline development/building-beagleboneblack.md]: Maybe we should remind the reader to move to the root of their home directory before cloning? `cd ~`

File: development/building-beagleboneblack.md (hunk 1 above)
Comment (redFrik) [reply development/building-beagleboneblack.md]: no there isn't.

Comment (patrickdupuis) [reply development/building-beagleboneblack.md]: my mistake :)

**Comment by redFrik** [comment]
thanks @patrickdupuis for suggestions and for trying it out. can someone merge this thanks. i'm not allowed to anymore.

**Comment by brianlheim** [comment]
Oh I guess patrick isn't part of core team yet so his approval doesn't allow merging. @patrickdupuis I just invited you to core team. Sorry, it shouldn't be this difficult to merge something, hopefully this helps. @redFrik I can also give you admin status on this repo if you want

**Pull request merged**